http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -386,29 +382,36 @@ class ClusterMonitorActor(
 
   private def createClusterInGoogle(
     cluster: Cluster
-  )(implicit ev: ApplicativeAsk[IO, TraceId]): Future[ClusterMonitorMessage] = {
-    val createClusterFuture = for {
-      _ <- IO(logger.info(s"Attempting to create cluster ${cluster.projectNameString} in Google...")).unsafeToFuture()
-      clusterResult <- clusterHelper.createCluster(cluster).unsafeToFuture()
-      now <- IO(Instant.now).unsafeToFuture()
-      (cluster, initBucket, saKey) = clusterResult
-      _ <- dbRef.inTransaction {
-        _.clusterQuery.updateAsyncClusterCreationFields(Some(GcsPath(initBucket, GcsObjectName(""))),
-                                                        saKey,
-                                                        cluster,
+  )(implicit ev: ApplicativeAsk[IO, TraceId]): IO[ClusterMonitorMessage] = {
+    val createCluster = for {
+      _ <- IO(logger.info(s"Attempting to create cluster ${cluster.projectNameString} in Google..."))
+      clusterResult <- clusterHelper.createCluster(cluster)
+      now <- IO(Instant.now)
+      _ <- dbRef.inTransactionIO {
+        _.clusterQuery.updateAsyncClusterCreationFields(Some(GcsPath(clusterResult.initBucket, GcsObjectName(""))),
+                                                        clusterResult.serviceAccountKey,
+                                                        clusterResult.cluster,
                                                         now)
       }
+      clusterImage = ClusterImage(ClusterImageType.CustomDataProc, clusterResult.customDataprocImage.asString, now)
+      // Save dataproc image in the database
+      _ <- dbRef.inTransactionIO(
+        _.clusterImageQuery.save(cluster.id, clusterImage)
+      ).handleErrorWith {
+        case e: java.sql.SQLIntegrityConstraintViolationException if(e.getMessage.contains("Duplicate entry")) => // This could happen when we recreate a cluster
```

Comment:
Catching the exception and pattern matching the message string may not be the best approach.. I feel like a SQL SELECT then INSERT if not present would be better.